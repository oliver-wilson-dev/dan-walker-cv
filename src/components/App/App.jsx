import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';

import Navigation from '../Navigation';
import HomePage from '../HomePage';
import AboutPage from '../AboutPage';
import ContactPage from '../ContactPage';
import Footer from '../Footer';

import styles from './App.module.css';

const App = () => (
  <Router>
    <div className={styles.App}>
      <Navigation />
      <Switch>
        <Route exact path="/">
          <HomePage />
        </Route>
        <Route path="/about">
          <AboutPage />
        </Route>
        <Route path="/contact">
          <ContactPage />
        </Route>
      </Switch>
      <Footer />
    </div>
  </Router>
);

export default App;
