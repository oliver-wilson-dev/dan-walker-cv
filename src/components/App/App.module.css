.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  justify-content: space-between;
}

a {
  text-decoration: none;
  color: rgb(21, 64, 204);
}

p {
  font-size: 14px;
}
.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
