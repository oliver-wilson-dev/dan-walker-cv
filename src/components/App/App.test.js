import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  BrowserRouter: ({ children }) => <>{children}</>,
  Route: ({ children }) => <>{children}</>,
  Switch: ({ children }) => <>{children}</>
}));

jest.mock('../Navigation', () => {
  const Navigation = () => null;

  return Navigation;
});

jest.mock('../HomePage', () => {
  const HomePage = () => null;

  return HomePage;
});

jest.mock('../AboutPage', () => {
  const AboutPage = () => null;

  return AboutPage;
});

jest.mock('../ContactPage', () => {
  const ContactPage = () => null;

  return ContactPage;
});

jest.mock('../Footer', () => {
  const Footer = () => <footer />;

  return Footer;
});

const render = () => shallow(<App />);

it('renders correctly', () => {
  expect(render()).toMatchSnapshot();
});
