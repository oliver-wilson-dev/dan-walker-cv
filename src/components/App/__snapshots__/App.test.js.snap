// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Navigation />
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <HomePage />
      </Route>
      <Route
        path="/about"
      >
        <AboutPage />
      </Route>
      <Route
        path="/contact"
      >
        <ContactPage />
      </Route>
    </Switch>
    <Footer />
  </div>
</BrowserRouter>
`;
